fn_cve_search/fn_cve_search/components/function_cve_search.py

```diff
@@ -0,0 +1,262 @@
+        elif isinstance(api_data, dict):
+            raise NotImplementedError("dictionary datatypes are not supported")
+        else:
+            raise NotImplementedError("result last cve call datatypes is not recognized")
+
+    def _parse_cve_results(self, api_data):
+        """
+
+        :param api_data: Specific CVE Rest API Call Result Data to Parsed
+        :return:an array of specific cve data
+        """
+        tmp_cve_list = []
+        # Changing The date format to milli seconds to store in the resilient table
+        search_pub_date = api_data.get('Published')
+        search_pub_date_timestamp = get_gm_epoch_time_stamp(date_string=search_pub_date)
+        api_data['Published'] = search_pub_date_timestamp
+        tmp_cve_list.append(api_data)
+        return tmp_cve_list
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_cve_search", {})
+
+    @function("function_cve_search")
+    def _function_cve_search_function(self, event, *args, **kwargs):
+        """Function: A Function to Search  Common Vulnerability Exposures Data from https://cve.circl.lu Data Base."""
+        try:
+            # Get the function parameters:
+            cve_search_data = kwargs.get("cve_search_data")  # text
+            if cve_search_data:
+                cve_search_data = cve_search_data.strip()
+
+            cve_search_criteria = kwargs.get("cve_search_criteria")  # text
+            if cve_search_criteria:
+                cve_search_criteria = cve_search_criteria.strip()
+
+            cve_id = kwargs.get("cve_id")  # text
+            if cve_id:
+                cve_id = cve_id.strip()
+
+            cve_vendor = kwargs.get("cve_vendor")  # text
+            if cve_vendor:
+                cve_vendor = cve_vendor.strip()
+
+            cve_product = kwargs.get("cve_product")  # text
+            if cve_product:
+                cve_product = cve_product.strip()
+
+            cve_published_date_from = kwargs.get("cve_published_date_from")  # datepicker
+            cve_published_date_to = kwargs.get("cve_published_date_to")  # datepicker
+
+            # Getting Max Result Display Flag from app.config
+            MAX_RESULTS_RETURN = int(self.options.get('max_results_display'))
+
+            # Variables to Store Parsed CVE API Data
+            _result_search_data = dict()
+
+            log = logging.getLogger(__name__)
+            log.info("cve_search_data: %s", cve_search_data)
+            log.info("cve_search_criteria: %s", cve_search_criteria)
+            log.info("cve_id: %s", cve_id)
+            log.info("cve_vendor: %s", cve_vendor)
+            log.info("cve_product: %s", cve_product)
+            log.info("cve_published_date_from: %s", cve_published_date_from)
+            log.info("cve_published_date_to: %s", cve_published_date_to)
+
+            yield StatusMessage("starting...")
```

Comment:
A better message like: "Searching the CVE Database. ID: {0}, Vendor: {1}......etc

Reply:
ok